src/yunohost/dyndns.py

```diff
@@ -252,6 +226,30 @@ def dyndns_update(dyn_host="dyndns.yunohost.org", domain=None, key=None,
         'zone %s' % host,
     ]
 
+
+    old_ipv4 = check_output("dig @91.224.148.92 +short %s" % domain).strip()
+    old_ipv6 = check_output("dig @91.224.148.92 +short aaaa %s" % domain).strip()
+
+    # Get current IPv4 and IPv6
+    ipv4_ = get_public_ip()
+    ipv6_ = get_public_ip(6)
+
+    if ipv4 is None:
+        ipv4 = ipv4_
+
+    if ipv6 is None:
+        ipv6 = ipv6_
+
+    logger.debug("Old IPv4/v6 are (%s, %s)" % (old_ipv4, old_ipv6))
+    logger.debug("Requested IPv4/v6 are (%s, %s)" % (ipv4, ipv6))
+
+    # no need to update
+    if old_ipv4 == ipv4 and old_ipv6 == ipv6:
```

Comment:
Need to check what happens here when, for instance, there's no IPv6. The previous `check_ouput()` probably returns empty string that is gonna be compared to `None`